core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataCreateIndexService.java

```diff
@@ -657,27 +637,81 @@ public void validateIndexSettings(String indexName, Settings settings) throws In
         return nodesToAllocateOn;
     }
 
-    static void prepareShrinkIndexSettings(ClusterState currentState, Set<String> mappingKeys, Settings.Builder indexSettingsBuilder, Index shrinkFromIndex, String shrinkIntoName) {
-        final IndexMetaData sourceMetaData = currentState.metaData().index(shrinkFromIndex.getName());
+    static void validateSplitIndex(ClusterState state, String sourceIndex,
+                                   Set<String> targetIndexMappingsTypes, String targetIndexName,
+                                   Settings targetIndexSettings) {
+        IndexMetaData sourceMetaData = validateResize(state, sourceIndex, targetIndexMappingsTypes, targetIndexName, targetIndexSettings);
+        IndexMetaData.selectSplitShard(0, sourceMetaData, IndexMetaData.INDEX_NUMBER_OF_SHARDS_SETTING.get(targetIndexSettings));
+        if (sourceMetaData.getCreationVersion().before(Version.V_6_0_0_alpha1)) {
+            // ensure we have a single type since this would make the splitting code considerably more complex
+            // and a 5.x index would not be splittable unless it has been shrunk before so rather opt out of the complexity
```

Comment:
can you add a note as to why? (I think it's routing shards but it's good not to have to wonder)